### PR TITLE
Feature/role back

### PR DESCRIPTION
**Pull request by AnastasiaLes**
1. Added logout 
2. Completed logic for user role

### CODE REVIEW

File: apps/freelance/src/pages/ChooseRolePage/index.tsx [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
 
 const { Text } = Typography;
 
 const ChooseRole = () => {
   const { t } = useTranslation();
-  const { handleSubmit } = useForm<Inputs>();
-  const [freelancer, setFreelancer] = useState(false);
+  const navigate = useNavigate();
+  const [role, setRole] = useState<IRole>({ role: 'Freelancer' });
+  const [freelancer, setFreelancer] = useState(true);
```
Comment (lyudad) [inline apps/freelance/src/pages/ChooseRolePage/index.tsx]: types

File: apps/freelance/src/pages/ChooseRolePage/index.tsx [hunk 2]
```diff
@@ -1,25 +1,28 @@
...
   const [jobOwner, setJobOwner] = useState(false);
```
Comment (lyudad) [inline apps/freelance/src/pages/ChooseRolePage/index.tsx]: types

File: apps/freelance/src/redux/services/types/user.types.ts [hunk 3]
```diff
@@ -34,3 +35,11 @@ export interface AddEducation {
   education_from: string;
   education_to: string;
 }
+
+export interface IUserInfo {
+  email: string;
+  first_name: string;
+  last_name: string;
+  profile_image: null;
```
Comment (lyudad) [inline apps/freelance/src/redux/services/types/user.types.ts]: its strange, if its always null, why its needed

File: apps/freelance/src/redux/services/types/user.types.ts [hunk 4]
```diff
@@ -34,3 +35,11 @@ export interface AddEducation {
...
+  role: 'Freelancer' | 'JobOwner';
```
Comment (lyudad) [inline apps/freelance/src/redux/services/types/user.types.ts]: 'Freelancer' | 'JobOwner' - make a separate type for it, you will reuse it often I think

File: apps/freelance/src/pages/ChooseRolePage/index.tsx (hunk 1 above)
Comment (AnastasiaLes) [reply apps/freelance/src/pages/ChooseRolePage/index.tsx]: added

File: apps/freelance/src/pages/ChooseRolePage/index.tsx (hunk 2 above)
Comment (AnastasiaLes) [reply apps/freelance/src/pages/ChooseRolePage/index.tsx]: added

File: apps/freelance/src/redux/services/types/user.types.ts (hunk 3 above)
Comment (AnastasiaLes) [reply apps/freelance/src/redux/services/types/user.types.ts]: fixed

File: apps/freelance/src/redux/services/types/user.types.ts (hunk 4 above)
Comment (AnastasiaLes) [reply apps/freelance/src/redux/services/types/user.types.ts]: Yes, thank you, made a separate type

File: apps/freelance/src/app/app.tsx [hunk 5]
```diff
@@ -43,118 +43,58 @@ export function App() {
...
-        <Route
-          path="/jobdetails"
-          element={
-            <PrivateRoute>
-              <JobDetailsPage />
-            </PrivateRoute>
-          }
-        />
+        <Route path="/" element={<Outlet />}>
+          <Route path="/welcome" element={<WelcomePage />} />
```
Comment (lyudad) [inline apps/freelance/src/app/app.tsx]: move all paths to constants, we often use them in other place. you can do it as separate task or pr but its good to have, while I remembered about it

File: apps/freelance/src/pages/ChooseRolePage/index.tsx [hunk 6]
```diff
@@ -1,25 +1,31 @@
...
 
 const { Text } = Typography;
 
 const ChooseRole = () => {
   const { t } = useTranslation();
-  const { handleSubmit } = useForm<Inputs>();
-  const [freelancer, setFreelancer] = useState(false);
-  const [jobOwner, setJobOwner] = useState(false);
+  const navigate = useNavigate();
+  const [role, setRole] = useState<IRole>({ role: null });
```
Comment (lyudad) [inline apps/freelance/src/pages/ChooseRolePage/index.tsx]: bad default state: or null or 0 or []

Comment (AnastasiaLes) [reply apps/freelance/src/pages/ChooseRolePage/index.tsx]: fixed
image

File: apps/freelance/src/app/app.tsx (hunk 5 above)
Comment (AnastasiaLes) [reply apps/freelance/src/app/app.tsx]: done

File: apps/freelance/src/pages/ChooseRolePage/index.tsx [hunk 7]
```diff
@@ -1,25 +1,34 @@
...
 
 const { Text } = Typography;
 
 const ChooseRole = () => {
   const { t } = useTranslation();
-  const { handleSubmit } = useForm<Inputs>();
-  const [freelancer, setFreelancer] = useState(false);
-  const [jobOwner, setJobOwner] = useState(false);
+  const navigate = useNavigate();
+  const [role, setRole] = useState<IRole>({ role: 'Visitor' });
```
Comment (lyudad) [inline apps/freelance/src/pages/ChooseRolePage/index.tsx]: Снимок экрана 2022-11-28 в 20 31 25

**Review by lyudad — APPROVED** [review]

**Pull request merged**